### ISSUE TITLE
GridOptionsView does not update properly when there are no items

### ISSUE THREAD

**Issue by OpticNectar** [UXDivers/Grial-UI-Kit-Support#309]
I have an `ObservableCollection` in an object bound to the`artina:GridOptionsView`. When I update the object the`artina:GridOptionsView` does not update properly if there are no items in the `ObservableCollection`. It works properly if there are one or more items though.

**View:**
```
<artina:GridOptionsView
    ColumnCount="{
        artina:OnOrientationInt 
            PortraitPhone=1,
            LandscapePhone=1,

            PortraitTablet=1,
            LandscapeTablet=1,
        
            PortraitDesktop=1,
            LandscapeDesktop=1
    }"
    ItemsSource="{ Binding MyObject.items }">
    <artina:GridOptionsView.ItemTemplate>
        <DataTemplate>
            <ctrls:MyItem />
        </DataTemplate>
    </artina:GridOptionsView.ItemTemplate>
</artina:GridOptionsView>
```

**Model:**
```
using System;
using System.Collections.ObjectModel;

namespace MyProject.Models
{
    public class MyObject : BaseModel
    {
      // name
      private string _name;
      public string name
      {
        get { return _name; }
        set { SetField(ref _name, value, "name"); }
      }

      // items
      private ObservableCollection<Item> _items;
      public ObservableCollection<Item> items
      {
        get { return _items; }
        set { SetField(ref _items, value, "items"); }
      }
    }
}
```

**ViewModel:**
```
private Order _Order;
public Order Order
{
  get { return _Order; }
  set { SetProperty(ref _Order, value); }
}
```

**Comment by OpticNectar** [comment]
Is there any way to get this fixed or a workaround? It is causing a lot of issues with my app.

**Comment by OpticNectar** [comment]
This seems to happen anytime the `artina:GridOptionsView` is set to an Observable collection that has items, and then the Observable collection is update to have no items. Please fix or give us a workaround ASAP. This is a HUGE issue and it's causing lots of problems with my app.

**Comment by OpticNectar** [comment]
@jpbrocca Can anyone help with this or give me a good work around? This is causing a delay in the app until we can get this issue fixed.

**Comment by dirivero** [comment]
Hi @OpticNectar.
Is this happening in Android or iOS? Does it throw an exception?

**Comment by OpticNectar** [comment]
@dirivero It's happening with both. 

To describe it a little better, basically I have a collection of objects in an `ObservableCollection` that i'm pulling from an API. The user is allowed to filter the results. If you filter the results and there are no objects in the `ObservableCollection`, then nothing happens and the previous items are still shown. If the results are more than 1, then they Update properly. I'm using the standard `Newtonsoft.Json` library if that helps at all.

**Comment by dirivero** [comment]
I've just tried that in both platforms and it seems to work:

![issue_309](https://user-images.githubusercontent.com/7670773/32754289-fe8b5880-c8ae-11e7-8e3a-3c6af21ba6a6.gif)

This just calls observableCollection.Clear() when CLEAR is pressed. Are you sure the collection is being cleared when there're no results in your filter?

Also, can you verify you have the latest version of our Artina nuget packages?

**Comment by OpticNectar** [comment]
@dirivero Try not calling `Clear()`, since that isn't what i'm doing. I'm just setting the collection to whatever is returned. So it would be setting it to and empty collection.

**Comment by dirivero** [comment]
Same result assigning an empty collection as long as you call the PropertyChanged event on the property that holds the collection after the assignment. That's needed since it's a brand new instance of an ObservableCollection, not the same that the GridOptionsView was observing.

Make sure the result is actually empty and that you are rising the PropertyChanged event.

If you want just send me the code so I can take a look.

**Comment by OpticNectar** [comment]
@dirivero I'm setting the `ObservableCollection` itself to the new result. When I output the `Count` it's zero, but the grid still isn't updating. I tried `RaisePropertyChanged` for the `ObservableCollection`, but that doesn't do anything and I shouldn't have to since i'm directly assigning it. It's weird cause I don't have this issue with `ListView` or other grids i've tried.

**Comment by OpticNectar** [comment]
@dirivero Let me know if you still can't replicate it and i'll build a sample project.

**Comment by dirivero** [comment]
Sadly no. I tried removing the .Clear() call, and instead assign an empty collection to the property in which the setter calls the `PropertyChanged` internally. I got the same result.

Please send me the sample code to drivero at uxdivers.com

**Comment by OpticNectar** [comment]
Is it drivero or dirivero?

**Comment by OpticNectar** [comment]
I sent it to both just to be sure. Let me know if you didn't get it.

**Comment by OpticNectar** [comment]
@dirivero Did you get the sample project in the email?

**Comment by dirivero** [comment]
Yes, I got it. I'll answer from there.

**Comment by LeoHere** [comment]
This issue was fixed starting at 2.5.14-RC3 available in the nuget package source. Closing the issue.
